File: python/run_analyses.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import pandas as pd
from pathlib import Path
import scipy.stats

# -----------------------------------------------------------------------------

data_dir = Path("..")

# -----------------------------------------------------------------------------

print("-" * 80)
print("Loading raw data...")
print()

data = pd.read_csv(data_dir / "simple_raw.csv", header=0)
print(data)
print()

# -----------------------------------------------------------------------------

print("-" * 80)
print("Processing data...")
print()

# Set the group/subject columns as the index
data = data.set_index(["group", "subject"])
# Only retain the numeric data
data = data.loc[:, "value"]

print(data)
print()

# -----------------------------------------------------------------------------

print("-" * 80)
print("Running statistics...")
print()

summary_stats = data.groupby(level=0).aggregate(["mean", "std"])
print(summary_stats)
print()

stats = scipy.stats.ttest_ind(
    data.loc["Thumb"], data.loc["Finger"], alternative="two-sided"
)
print(stats)
print()

# -----------------------------------------------------------------------------

print("-" * 80)

# -----------------------------------------------------------------------------

File: R/run_analyses.R
#!/usr/bin/env Rscript

library(tidyverse)

cat_line <- function(...) cat(..., "\n", sep = "")

# -----------------------------------------------------------------------------

data_dir <- ".."

# -----------------------------------------------------------------------------

cat_line(rep("-", 80), collapse = "")
cat_line("Loading raw data...")
cat_line()

data <- read_csv(
    file.path(data_dir, "simple_raw.csv"),
    col_names = TRUE,
    show_col_types = FALSE
)
print(data)
cat_line()

# -----------------------------------------------------------------------------

cat_line(rep("-", 80), collapse = "")
cat_line("Processing data...")
cat_line()

# Only retain the group/subject columns and the numeric data
data <- data %>%
    select(group, subject, value)

print(data)
cat_line()

# -----------------------------------------------------------------------------

cat_line(rep("-", 80), collapse = "")
cat_line("Running statistics...")
cat_line()

summary_stats <- data %>%
    group_by(group) %>%
    summarise(mean = mean(value), sd = sd(value))
print(summary_stats)
cat_line()

stats <- t.test(data$value ~ data$group, alternative = "two.sided")
print(stats)

# -----------------------------------------------------------------------------

cat_line(rep("-", 80), collapse = "")

# -----------------------------------------------------------------------------
